setup.cfg

```diff
@@ -54,6 +54,7 @@ classifiers =
 include_package_data = True
 install_requires =
   importlib-metadata >= 4; python_version < '3.8'
+  dataclasses; python_version < '3.7'
```

Comment:
I don't think that it makes sense to drag a whole dependency in, for the case that never needs it. The Towncrier version that uses `dataclasses` does not support Python 3.6, so we know for sure that `dataclasses.is_dataclass(config)` will always be `False` under `python_version < '3.7'` and so `dataclasses.asdict(config)` will never be called when the backport is installed.